feat: manage parameters and URL

adschwartz opened the pull request:
## Description:
Changes:
* Accessing the package config from anywhere in the UI
* Integration with URL parameters in the EM UI and the Cloud UI


## Is this change user facing?
YES

Commit:
refactor creation and URL loading

Commit:
create-enclave works pretty well

Commit:
misc

Commit:
get those urls straightened out to work with cloud ui

Commit:
simple build script

Commit:
build

Commit:
disable

Commit:
reset

Commit:
new build

Commit:
Merge branch 'main' into anders/parameters

# Conflicts:
#	enclave-manager/web/src/components/enclaves/modals/ConfigureEnclaveModal.tsx

Dartoxian commented on enclave-manager/web/src/client/enclaveManager/KurtosisClient.ts:
```diff
@@ -73,11 +79,11 @@ export abstract class KurtosisClient {
       const request = new GetListFilesArtifactNamesAndUuidsRequest({
         apicIpAddress: apicInfo.bridgeIpAddress,
-        apicPort: apicInfo.grpcPortInsideEnclave,
+        apicPort: apicInfo.grpcPortInsideEnclave
```
wonder if our prettiers aren't configured consistently - this change is just a formatting change. Could you run `yarn prettier:fix` and verify whether you keep this change.

adschwartz replied:
indeed, my client is not picking up the `prettier` formatter. Will fix that

adschwartz replied:
We should add an ignore for protobuf generated files as it also formats those guys.

adschwartz replied:
done

Dartoxian commented on enclave-manager/web/src/components/constants.ts:
```diff
@@ -0,0 +1,5 @@
+export const KURTOSIS_PACKAGE_ID_URL_ARG = "package-id"
+export const KURTOSIS_CREATE_ENCLAVE_URL_ARG = "create-enclave"
+
+export const KURTOSIS_PACKAGE_URL_BASE="/"
```
I'd prefer to have constants defined near to where they are used (ie components/mo- arguably having a `constants` file in a `components/enclaves/modals`) directory is a bad smell anyway - it suggests we've started working quite a lot of application logic into the components (which tbf is what I've given you...)

Dartoxian commented on enclave-manager/web/src/components/enclaves/CreateEnclave.tsx:
```diff
@@ -0,0 +1,58 @@
+  return (
+    <>
+      <PreloadPackage onPackageLoaded={handleOnPackageLoaded} />
+      <ManualCreateEnclaveModal
+        isOpen={manualCreateEnclaveOpen}
+        onClose={handleCloseManualCreateEnclave}
+        onConfirm={handleManualCreateEnclaveConfirmed}
+      />
```
took me a moment to work out how you're getting this open now, but makes sense. In a non SPA I think I'd not be that comfortable driving state changes like this, but for this application where we're relying a lot on react-router-dom to manage state changes this does seem more appropriate.

adschwartz replied:
I'm using the url fragment to trigger it to open from any "page". 

Would be (genuinely) interested to hear what you'd propose alternatively and what specifically would not make this a good fit in a not SPA?

Dartoxian replied:
One alternative would be to drive it entirely from react state/structure, which is what I was doing before (and is perhaps slightly more readable as it's explicit how it's used as you read the react tree) - but this does seem preferable in terms of getting the behaviour easily anywhere in the app.

The 'not a SPA' example I am thinking about is anything that uses a server rendering framework (specifically thinking nextjs), in those frameworks using effects to render content is generally avoided - and the page is hydrated from the server with relevant data if possible - so we'd probably prefer to use a configuration page, rather than a modal.

adschwartz replied:
Cool! This all makes sense. Thanks!

Dartoxian commented on enclave-manager/web/src/components/enclaves/configuration/EnclaveConfigurationForm.tsx:
```diff
@@ -98,7 +98,7 @@ export const EnclaveConfigurationForm = forwardRef<
       );
 
-    onSubmit({ ...data, args: newArgs });
+    onSubmit({enclaveName: "", restartServices: false, ...data, args: newArgs });
```
I am not sure this is needed - these fields should already be supplied by `ConfigureEnclaveModal` (ie they should never be `null` or undefined in the supplied `data` object)

adschwartz replied:
yeah this was a fix to get it to compile. Wondering what data we are missing to wire in here.

adschwartz replied:
I can't get this to compile if I don't add those members, is there something off in my config? How are you getting this to compile? It makes sense that the descruturing should provide this data but running yarn build fails on mine, so wondering what you have that works?
![image](https://github.com/kurtosis-tech/kurtosis/assets/11248996/c0e1e321-affd-4e71-acaf-a07100e00960)

Dartoxian replied:
interesting, no I don't get this same issue when I run yarn run build.

In the interest of expediency, I'm not particularly interested in working out what the difference is atm.

Dartoxian reviewed (commented):
Main points:
* please check prettier - if you want to change the rules could you also commit them. Consistent code formatters reduces noise in code review.
* Given you've gone part way to adding the build change could you also .gitignore the engine output?

Dartoxian commented on enclave-manager/web/src/components/CopyButton.tsx:
```diff
@@ -15,7 +15,7 @@ export const CopyButton = ({ valueToCopy, text, ...buttonProps }: CopyButtonProp
       navigator.clipboard.writeText(v);
       toast({
-        title: `Copied '${v}' to the clipboard`,
+        title: `Copied url to the clipboard`,
```
It's not always a url that is copied - I think this is better exposed as a prop.

adschwartz replied:
Done

Commit:
new build + prettier formatting

Commit:
Merge branch 'main' into anders/parameters

Commit:
changes based on feedback

Commit:
more fixes

Commit:
Work with relative base

Commit:
new build

Commit:
Merge branch 'main' into anders/parameters

Commit:
reset

Commit:
Merge branch 'main' into anders/parameters

Commit:
update

Dartoxian commented on enclave-manager/web/src/client/enclaveManager/KurtosisClientContext.tsx:
```diff
@@ -131,3 +138,12 @@ export const useKurtosisClient = (): KurtosisClient => {
+
+const paramToUrl = (searchParams: URLSearchParams, param: string) => {
+  let paramString = searchParams.get(param);
+  if (paramString === null) return null;
```
I'll add eslint config for `curly` -https://eslint.org/docs/latest/rules/curly. Generally I dislike this form of conditional as it's less readable and a source of developer errors. Famous example of why not to do this https://www.synopsys.com/blogs/software-security/understanding-apple-goto-fail-vulnerability-2.html

adschwartz replied:
That's fair. Thanks for fixing that!

Dartoxian reviewed (commented):
I think prettier:fix also needs to be run over this change.

Dartoxian committed:
Prettier and prettier-curly

Dartoxian commented:
> I think prettier:fix also needs to be run over this change.

done - and added curly

Dartoxian committed:
Merge branch 'main' into anders/parameters

Dartoxian committed:
Update build

Commit:
simplify

Commit:
clean ip

Commit:
clean

Dartoxian committed:
feat: new emui service logs (#1702)

## Description:
This change adds support for showing enclave service logs in the new
enclave manager ui.

Additionally, this pr contains a partial implementation of the service
view ui - full implementation to follow.

### Demo:

In this demo `github.com/kurtosis-tech/log-load-package` is used to
demonstrate large quantities of logs don't seem to break following.
Additionally, my browser has it's locale set to san francisco - so that
browser locale dependent timestamps can be demonstrated.


https://github.com/kurtosis-tech/kurtosis/assets/4419574/533b5f5a-7fdf-40b1-a6c5-e0b67970811c

## Is this change user facing?
YES
